### ISSUE TITLE
Discrepancy in categorical_crossentropy between evaluate and predict

### ISSUE THREAD

**Issue by henry-eigen** [keras-team/keras#14003]
**System information**  
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  2.0.0
- Keras version:  2.3.1
- Python version:  3.6.8

**Describe the current behavior**  

The loss given by model.evaluate using categorical_crossentropy as the loss function is different than the loss given by computing categorical_crossentropy with the output of model.predict as y_pred and the labels as y_true.

**Code to reproduce the issue**  
```
model.compile(loss='categorical_crossentropy', optimizer=Adam(lr=1e-4))
model.evaluate(x_test[:16], y_test[:16])  

# Gives loss of 0.2167

preds = model.predict(x_test[:16])
K.mean(tf.keras.losses.categorical_crossentropy(y_test[:16], preds))

# Gives loss of 0.0557
```

**Comment by biddwan09** [comment]
Hi I am very interested to contribute in Keras. Can I look into this issue

**Comment by henry-eigen** [comment]
@biddwan09 It looks like no-one else is so please go ahead. I might check to see if this issue is still present with TF 2.2+ first, but do report back if you find anything.

**Comment by tilakrayal** [comment]
@henry-eigen ,
Can you please try with latest versions and let us know if the issue still persists.Thanks!
